sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/higherOrderFunctions.scala

```diff
@@ -155,6 +155,8 @@ trait HigherOrderFunction extends Expression with ExpectsInputTypes {
  */
 trait SimpleHigherOrderFunction extends HigherOrderFunction  {
 
+  override def nullable: Boolean = argument.nullable
```

Comment:
If we moved the definition of ```nullable``` straight to ```HigherOrderFunction``` as ```arguments.exists(_.nullable)```, we could also avoid the duplicities in ```ZipWith``` and ```MapZipWith```. WDYT?

Reply:
this works too IMO, if others agree I'll update with this suggestion, thanks.

Reply:
Yea, let's go ahead then if the change is small, straightforward and more deduplciation